### PR TITLE
feat: Add getting started ressources section to Homepage, project pages and projects page

### PR DESCRIPTION
**Pull request by blushi**
## Description

Closes: regen-network/regen-registry#1372

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] provided a link to the relevant issue or specification
- [x] provided instructions on how to test
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed
- [ ] once the PR is closed, set up backport PRs for `redwood`, `hambach` and optionally `master` (see below)

#### Setting up backport PRs

After merging your PR to `dev`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR:

- If PR is a new feature: `@Mergifyio backport redwood hambach`.
- If PR is a bug fix: `@Mergifyio backport redwood hambach master`.

2. If your branch does have merge commits:

   a. Pull latest `dev`, `hambach` and `redwood` branches

   b. Create new branches for backports and merge `dev` (replace `<PR#>` with your PR #)

   ```
   git checkout -b hambach-backport-<PR#> hambach
   git merge dev
   git push origin hambach-backport-<PR#>
   git checkout -b redwood-backport-<PR#> redwood
   git merge dev
   git push origin redwood-backport-<PR#>`
   ```

   c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.

   d. If PR is a bug fix repeat the same process for `master` branch.

### How to test

Go to https://deploy-preview-1613--regen-registry.netlify.app/
and check out home page, /projects page and a project page

Content on the bridge page has also been updated to pull from sanity instead of being harcoded.

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-components/src/components/section/SliderSection.tsx [hunk 1]
```diff
@@ -49,6 +50,8 @@ const useStyles = makeStyles()((theme: Theme) => ({
 function SliderSection({
   items,
   title,
+  slidesToShow,
+  // titleAlign = 'left',
```
Comment (blushi) [inline web-components/src/components/section/SliderSection.tsx:54]:
```suggestion
```

**Review by mhagel — COMMENTED** [review]
Mobile seems a bit odd. It lets the whole page scroll left:


Screen Shot 2022-11-29 at 5 05 27 PM

**Comment by flagrede** [comment]
> > Mobile seems a bit odd. It lets the whole page scroll left:
> 
> weird, but somehow I can't reproduce it on a real device, wondering if that has something to do with the devtools

@blushi I managed to reproduce it with both dev tools and a real device. 
The issue seems to be that the new resources section container is overflowing:
image
An `overflow-x: hidden` on the container should fix it.

**Comment by blushi** [comment]
thanks @flagrede but it seems like we do need this overflow on larger screens on the container, otherwise with >= 3 cards, it looks like this:

image
 
I'll update the code accordingly.

File: web-registry/src/components/molecules/GettingStartedResourcesSection.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+import { Link } from 'components/atoms';
+
+const GettingStartedResourcesSection: React.FC<
+  React.PropsWithChildren<{
+    section: GettingStartedResourcesSectionFieldsFragment;
+  }>
+> = ({ section }) => {
+  const theme = useTheme<Theme>();
+  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
```
Comment (flagrede) [inline web-registry/src/components/molecules/GettingStartedResourcesSection.tsx:22]: Here we don't have the choice but to do a client check I think.

**Review by mhagel — APPROVED** [review]
Thanks!

**Comment by blushi** [comment]
@erikalogie could you review this please?

File: web-components/src/components/cards/GettingStartedResourcesCard/GettingStartedResourcesCard.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+import { Theme } from '../../../theme/muiTheme';
+import { BlockContent, SanityBlockContent } from '../../block-content';
+import SmallArrowIcon from '../../icons/SmallArrowIcon';
+import { LinkComponentProp } from '../../modal/ConfirmModal';
+import { Body, Label } from '../../typography';
+import { Title } from '../../typography/Title';
+import Card from '../Card';
+
+export interface GettingStartedResourceCardProps {
+  header?: string | null;
```
[the remark this answers is not in the document]
Comment (blushi) [reply web-components/src/components/cards/GettingStartedResourcesCard/GettingStartedResourcesCard.tsx:12]: yeah this is coming from the generated sanity types
it's either we do that or use `header={header || ''}` (which we do in some places already but not so fond of it)

File: web-components/src/components/cards/GettingStartedResourcesCard/GettingStartedResourcesCard.tsx [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+  header,
+  description,
+  imageUrl,
+  mobileImageUrl,
+  links,
+  linkComponent: LinkComponent,
+  fullWidth = false,
+}: GettingStartedResourceCardProps): JSX.Element => {
+  const theme = useTheme();
+  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
```
[the remark this answers is not in the document]
Comment (blushi) [reply web-components/src/components/cards/GettingStartedResourcesCard/GettingStartedResourcesCard.tsx]: good point! we need to use `sm: ...` though, updated

File: web-registry/src/components/molecules/GettingStartedResourcesSection.tsx (hunk 2 above)
Comment (blushi) [reply web-registry/src/components/molecules/GettingStartedResourcesSection.tsx:22]: indeed

**Comment by erikalogie** [comment]
@blushi this looks good, just some small nits:

Mobile related: https://monosnap.com/file/XREwg5c7WIRAe3eRibrEuaOh5aM89N
Desktop related: https://monosnap.com/file/DYLMrM7ekFwRvFBLNLbL25gT9fYRWi This might be better to fix as part of this issue: https://app.zenhub.com/workspaces/regen-registry-5f8998bec8958d000f4609e2/issues/regen-network/regen-registry/1374 as the stroke I think isn't necessarily part of this component. The linked issue also includes fixing the "more projects" component on the individual project pages so that it has the topo background and the stroke top and bottom. I think we should do this next sprint because now it looks even weirder to not have this separation with the Resources section added on the individual project pages.

**Comment by blushi** [comment]
@erikalogie 

> Mobile related: https://monosnap.com/file/XREwg5c7WIRAe3eRibrEuaOh5aM89N

fixed

> Desktop related: https://monosnap.com/file/DYLMrM7ekFwRvFBLNLbL25gT9fYRWi This might be better to fix as part of this issue: https://app.zenhub.com/workspaces/regen-registry-5f8998bec8958d000f4609e2/issues/regen-network/regen-registry/1374 

let's address as part of regen-network/regen-registry#1374

**Comment by erikalogie** [comment]
Looks great!

**Comment by blushi** [comment]
@Mergifyio backport redwood hambach

**Pull request merged**